addons/csi/azure-file/Makefile

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 The Kubermatic Kubernetes Platform contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/charts
+
+OUTPUT_FILE = driver.yaml
+REPO_NAME = kkp-addons-csi-azure-file
+CHART_VERSION = 1.30.2
+
+.PHONY: default
+default: setup-helm build clean-helm
+
+.PHONY: build
+build:
+	helm --namespace kube-system template azurefile-csi-driver $(REPO_NAME)/azurefile-csi-driver \
+	  --version $(CHART_VERSION) \
+	  --set 'windows.enabled=false' \
+	  --set 'controller.replicas=1' \
+	  --set 'controller.podAnnotations.cluster-autoscaler\.kubernetes\.io/safe-to-evict-local-volumes=socket-dir' \
+	  --skip-tests \
+	  > $(OUTPUT_FILE)
```

Comment:
q: Is there a reason why we cannot use kustomizes helm feature here? If so, I think we should document it

Reply:
I consider Helm to be the source and Kustomize to be one of the transformation steps, hence why I separated the two. Personally I see no reason to intermingle Helm rendering and kustomizing in the first place, that Helm integration smells like feature creep to me. ^^

Reply:
hmm interesting. For me it's the opposite: I consider kustomize the source of truth, especially with a pinned helm-chart version. It's the higher-power renderer responsible for producing a predictable result.
But tbf, this is a tomatos <> tomatoes kind of thing for me. Will approve the PR now